woocommerce/woocommerce
# Enhance order details and payment summary

### Changes proposed in this Pull Request:

This pull request makes incremental enhancements to the order details screen. This update also better aligns the order details and payment summary with the way we display the same information in the WooCommerce mobile application.

1. It changes the **cost** of a line item to display it's GROSS amount (before any discounts are applied) This makes it easier to understand how a discount is being applied on a line by line basis.

**Before:**

<img width="868" alt="Screen Shot 2020-01-27 at 7 53 23 PM" src="https://user-images.githubusercontent.com/4500952/73234433-eee53d00-413e-11ea-8983-97e37c96ff19.png">

**After:**

<img width="869" alt="Screen Shot 2020-01-27 at 7 54 02 PM" src="https://user-images.githubusercontent.com/4500952/73234694-17216b80-4140-11ea-8260-f0d5a685a215.png">

2. It adds critical, missing information to the summary section of the order details

- A subtotal of the items
- Fees
- Once the order status becomes "Processing" it displays "Paid by customer" with the amount paid on a new line below the order total

3. It better clarifies new order totals in the case of refunds:

- It displays the total amount refunded below "Paid by customer" along with a new net payment amount

**Before & after**

<img width="2646" alt="Screen Shot 2020-01-27 at 8 17 59 PM" src="https://user-images.githubusercontent.com/4500952/73235255-813b1000-4142-11ea-9d20-421794534136.png">

**Before & after with a refund**

<img width="2644" alt="Screen Shot 2020-01-27 at 9 13 09 PM" src="https://user-images.githubusercontent.com/4500952/73237284-e8a88e00-4149-11ea-89e5-411a446ffe18.png">


### How to test the changes in this Pull Request:

1. Load up this branch
2. Create a new order from scratch - add all the things. Discounts, fees, shipping, taxes
3. Verify all of the lines in the summary are correct, and that their sum matches the order total
4. Pay for the order (the status should update to "Processing")
5. Verify the summary now says "Paid by customer" along with the amount paid (The order total)
6. Refund the order
7. Verify the amount paid minus the refunded amount equals the amount displayed beside "net total"

### Needs testing with additional gateways

I've tested this with the Stripe plugin for WooCommerce, however it would be worth testing other popular gateways to ensure any extra information added by a gateway isn't throwing the summary list out of order. 

In the screen shot below, you'll see that the stripe gateway adds the transaction fee, and the payout amount, however, they're positioned at the bottom of the summary, so the user can simply continue to sum down the column as expected. 

<img width="1434" alt="Screen Shot 2020-01-27 at 9 14 21 PM" src="https://user-images.githubusercontent.com/4500952/73237321-0544c600-414a-11ea-829d-85dbc360e68b.png">


### Other information:




### Changelog entry

> Update line item cost to display gross amount. Add items subtotal and fees to order summary, along with "Paid by Customer" once the order status changes to processing
